dnplab/dnpImport/prospa.py

```diff
@@ -239,3 +228,62 @@ def import_csv(path, return_raw=False, is_complex=True):
         return x, data
     else:
         return raw
+
+
+def prospa_coords(attrs, data_shape):
+    """Generate coords from prospa acquisition parameters"""
+
+    experiment = attrs["experiment"]
+    dims = []
+    coords = []
+
+    if experiment == "1Pulse":
+        pts = attrs["nrPnts"]
+        dwell_time = attrs["dwellTime"]
+        x = np.arange(0.0, pts * dwell_time, dwell_time) / 1e6
+        dims.append("t2")
+        coords.append(x)
+
+    elif experiment == "1Pulse2dAverage":
+        pts = attrs["nrPnts"]
+        dwell_time = attrs["dwellTime"]
+        x = np.arange(0.0, pts * dwell_time, dwell_time) / 1e6
+        dims.append("t2")
+        coords.append(x)
+
+        dims.append("Average")
+        coords.append(np.arange(attrs["nrScans"]))
+
+    elif experiment == "T1-IR-FID":
+        pts = attrs["nrPnts"]
+        dwell_time = attrs["dwellTime"]
+        x = np.arange(0.0, pts * dwell_time, dwell_time) / 1e6
+        dims.append("t2")
+        coords.append(x)
+
+        T1_steps = attrs["nrSteps"]
+        T1_min_delay = attrs["minDelay"]
+        T1_max_delay = attrs["maxDelay"]
+
+        if attrs["delaySpacing"] == "lin":
+            T1 = np.linspace(T1_min_delay, T1_max_delay, T1_steps) / 1000.0
+        else:
+            T1 = np.logspace(T1_min_delay, T1_max_delay, T1_steps) / 1000.0
+
+        dims.append("t1")
+        coords.append(T1)
+    else:
+        dims_list = ["x", "y", "z", "q"]
+        for ix in range(len(data_shape)):
+            dims.append(dims_list[ix])  # call dimensions in order: x, y, z, q
```

Comment:
This is confusing to me. You need to assert `len(dims_list) >= len(data_shape)` ? Is it true?

Reply:
If I cannot determine the coords from the acquisition parameters file, this section of code will create coords based on the index.

Reply:
> If I cannot determine the coords from the acquisition parameters file, this section of code will create coords based on the index.

My question is what if `len(data_shape) > len(dims_list)` ?